### ISSUE TITLE
When title in metadata, can not search and show title in search list.

### ISSUE THREAD

**Issue by gledos** [squidfunk/mkdocs-material#3980]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

When title in metadata, can not search and show title in search list.

### Expected behaviour

Title in search list.

### Actual behaviour

Can not search and show title in search list.

Discussions: https://github.com/squidfunk/mkdocs-material/discussions/3978

### Steps to reproduce

Markdown file like:

```markdown
---
title: test
---

## H2
```

### Package versions

- Python: `3.9.7`
- MkDocs: `1.3.0`
- Material: `8.2.7`

### Configuration

```yaml
site_name: metadata title test

theme:
  name: material

plugins:
  - search
```

### System information

- Operating system: Windows 10
- Browser: Firefox 100

**Comment by squidfunk** [comment]
Confirmed. MkDocs will not respect the `title` via front matter when building the search index. Also, in your example, you did not explicitly set the `h1`, which will result in the first `h2` being the page title, which is already fixed in the [new built-in search plugin](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/#built-in-search-plugin), a complete rewrite of MkDocs own search plugin.

**Comment by squidfunk** [comment]
Fixed in [`0f19611`](https://github.com/squidfunk/mkdocs-material-insiders/commit/0f19611bac31f1ce67e9a27188fd406b851328fa). The new built-in search will now use the `title` set via front-matter over the `h1` or the page name in `mkdocs.yml`. Note that we can only ship this fix as part of Insiders, as the new built-in search was not released into the community edition yet. In case you don't want to use Insiders, please upstream this issue to [MkDocs](https://github.com/mkdocs/mkdocs).

**Comment by gledos** [comment]
thanks for your help, I have submitted this issue to MkDocs.

https://github.com/mkdocs/mkdocs/issues/2859

So this issue should be closed now, right?

**Comment by squidfunk** [comment]
I'll close it when I released the changes in Insiders 😊

**Comment by squidfunk** [comment]
Released as part of 8.3.1+insiders-4.17.0.
